src/js_native_api_types.h

```diff
@@ -20,6 +20,13 @@ typedef struct napi_handle_scope__* napi_handle_scope;
 typedef struct napi_escapable_handle_scope__* napi_escapable_handle_scope;
 typedef struct napi_callback_info__* napi_callback_info;
 typedef struct napi_deferred__* napi_deferred;
+#define node_api_env napi_env
```

Comment:
I think all of these defines need to be guarded by NODE-API version 8. Otherwise add-ons could fail to compile against versions of Node.js that support the NODE-API version they are targeting but don't have the new defines. I see adding these defines as the equivalent to adding new supported functions to NODE-API.

Reply:
@mhdawson one concern is that `NAPI_VERSION` itself has to change so, going forward, people would be able to select API versions by defining NODE_API_VERSION themselves.